File: src/javaapp/DummyDataFactory.java
package javaapp;

import java.util.ArrayList;
import java.util.List;

public class DummyDataFactory {

    public static List<Product> createDummyData() {
        List<Product> products = new ArrayList<>();
        products.add(new Product("Laptop", 700, 10));
        products.add(new Product("Phone", 500, 5));
        products.add(new Product("Headphones", 100, 0));
        products.add(new Product("Mouse", 30, 20));
        products.add(new Product("Keyboard", 50, 15));
        return products;
    }

    public static void preloadData(ProductCatalog productCatalog) {
        productCatalog.products.addAll(createDummyData());
    }
}

File: src/kotlinapp/DummyDataFactory.kt
package kotlinapp

import javaapp.Product
import javaapp.ProductCatalog
import java.util.ArrayList

object DummyDataFactory {
    fun createDummyData(): List<Product> {
        val products: MutableList<Product> = ArrayList()
        products.add(Product("Laptop", 700f, 10))
        products.add(Product("Phone", 500f, 5))
        products.add(Product("Headphones", 100f, 0))
        products.add(Product("Mouse", 30f, 20))
        products.add(Product("Keyboard", 50f, 15))
        return products
    }

    fun preloadData(productCatalog: ProductCatalog) {
        productCatalog.products.addAll(createDummyData())
    }
}